python/common/org/python/types/Str.java

```diff
@@ -708,10 +708,21 @@ private long toPositiveIndex(long index) {
     }
 
     @org.python.Method(
-            __doc__ = ""
+            __doc__ = "string.isalnum() -> bool\n" +
+                    "\n" +
+                    "Returns false when the string has atleast ONE character which is\n" +
+                    "not an alphanumeric or if the string is empty.\n"
     )
     public org.python.Object isalnum() {
-        throw new org.python.exceptions.NotImplementedError("isalnum() has not been implemented.");
+        if (!this.value.isEmpty()) {
```

Comment:
Sure, I understand. It's certainly a very good idea to think twice about whether there's some reason behind what may initially appear to be strange/weird code!

Sometimes what I do when I'm not sure is regardless of whether I follow or change it to a different way, I highlight it in the message/comments when creating a PR, so that anyone else can also take a closer look and agree/disagree with the action. It could be an opportunity to improve both the current PR and existing code!